commit/merkleroot/rmn/controller.go

```diff
@@ -310,12 +312,26 @@ func (c *controller) sendObservationRequests(
 			return requests[i].LaneSource.SourceChainSelector < requests[j].LaneSource.SourceChainSelector
 		})
 
+		fixedDestLaneUpdateRequests := make([]*rmnpb.FixedDestLaneUpdateRequest, 0)
+
+		// convert OnRamp address from 32bytes (abi encoded) to 20bytes (evm address) before sending the request
+		// todo: make this part chain-agnostic
+		for _, request := range requests {
+			fixedDestLaneUpdateRequests = append(fixedDestLaneUpdateRequests, &rmnpb.FixedDestLaneUpdateRequest{
+				LaneSource: &rmnpb.LaneSource{
+					SourceChainSelector: request.LaneSource.SourceChainSelector,
+					OnrampAddress:       typconv.KeepNRightBytes(request.LaneSource.OnrampAddress, 20),
```

Comment:
If we want this to be more general (at least on an interface level) maybe this function can be something like 

```golang
func BytesToAddress(bytesAddress []byte, chainSel uint64) ([]byte, error) {
  family, err := chainsel.GetSelectorFamily(chainSel)
  if err != nil { return nil, fmt.Errorf("get selector family: %w", err) }
  switch family {
    case chainsel.FamilyEVM:
      return bytesAddress[uint(len(b))-n:]
    case chainsel.FamilySol:
      // ...
    default:
      return nil, fmt.Errorf("unknown family: %s", family)
  }
}
```

This API is actually available right now: https://github.com/smartcontractkit/chain-selectors/blob/5b55b2f9e17e46e14481baa94eef8c33a6998b1f/selectors.go#L90. I think we should start cutting over to it in the chainlink-ccip codebase.